### ISSUE TITLE
Training on single channel 16 bit images

### ISSUE THREAD

**Issue by Lakshya-Kejriwal** [fizyr/keras-retinanet#483]
Hi, 

I am wondering if it is possible to use retinanet to train for 16 bit single channel TIFF format images. 
Where is the best place to start for making modification of the code?

**Comment by yhenon** [comment]
I've used retinanet with grayscale medical images that were 16-bit in the range 0-65535.
- if your images are in the range 0-65535 (as can be the case with 16-bit images), you might want to change how normalization is done in `keras_retinanet/utils/image.py`, since subtracting imagenet means doesn't make sense. In my case I divided by 65535 and subtracted 0.5, which worked well.
- converting from grayscale to RGB makes things easy and allows you to use the pretrained imagenet weights.
- If PIL can't handle your image, you might have to edit `read_image_bgr` in `keras_retinanet/utils/image.py`.

**Comment by Lakshya-Kejriwal** [comment]
So if I don't convert from grayscale to RGB, I would need to start training with no weights. Is that correct? And if I do that then I won't need to pre-process the image?

**Comment by yhenon** [comment]
>So if I don't convert from grayscale to RGB, I would need to start
training with no weights. Is that correct
Well you could try with imagenet weights, except for the first layer.
There's no real benefit, except for the small computation gain from having
1 channel instead of 3.
> And if I do that then I won't need to pre-process the image?

You'll still need to convert your image to 32-bit of course, and you might
still need to normalize it. You'd also have to change the backbone to use
only 1 input channel.

This conversation is better suited for slack, please see README for
instructions on using the slack.
